packages/admin/cms-admin/src/blocks/ExternalLinkBlock.tsx

```diff
@@ -42,6 +42,17 @@ export const ExternalLinkBlock: BlockInterface<ExternalLinkBlockData, State, Ext
         return state.targetUrl ? isHref(state.targetUrl) : true;
     },
 
+    fromRaw: (raw) => {
```

Comment:
This function is *very* specific, I would change it to `fromLinkTarget(url: string)` or something like that.

Reply:
and add 2State (or ToState) as it returns state

Reply:
Don't you think that `linkUrl2State` would be too specific for our generic `BlockInterface`? I thought this method could be used for future use cases as well, for instance creating a rich text block state from a plain string

Reply:
True. Why not create a dedicated `LinkBlockInterface`? Or simply omit it from the interface?

Reply:
Because our link block are usually `OneOfBlock`s

> Or simply omit it from the interface?

I don't understand that. Would I simply call the function on the block, even if it doesn't exist in the interface?

Reply:
> Because our link block are usually OneOfBlocks

yes, a bit strange if all OneOf implement it. A subclass (usage with createLinkOneOf or so) could work. Similar situation you have for redirect links, or was this on the api side only?

> I don't understand that. Would I simply call the function on the block, even if it doesn't exist in the interface?

yes, in javascript you can do that :) (yes, it's not pretty and not properly typed - but it's nothing application code has to do)

Reply:
> A subclass (usage with createLinkOneOf or so) could work

I've now added a `LinkBlockInterface` interface and a `createLinkBlock` factory that supports `url2State`: f5d4b2d88396d939aae3cb516df212d7793e793a

> Similar situation you have for redirect links, or was this on the api side only?

It's a bit different with the redirects: Here the custom supported blocks are passed as options, and the one of block is created by the redirects module itself.